### DISCUSSION TITLE
wrangler pages deploy - source maps

### DISCUSSION THREAD

**Discussion by BrandonNoad** [cloudflare/workers-sdk#4234] [Ideas]
Would it be possible to get access to the source maps for the `functionsWorker` files that are built and deployed when using `wrangler pages deploy`?

Currently, our error logs look like this:

```
TypeError: Unrecognized Content-Type header value. FormData can only parse the following MIME types: multipart/form-data, application/x-www-form-urlencoded
    at Object.validateToken (functionsWorker-0.10525428848427065.js:49066:14)
    at Upt (functionsWorker-0.10525428848427065.js:49116:3)
    at SS (functionsWorker-0.10525428848427065.js:2151:11)
    at J0 (functionsWorker-0.10525428848427065.js:1581:11)
    at s (functionsWorker-0.10525428848427065.js:1383:20)
    at c (functionsWorker-0.10525428848427065.js:1362:16)
    at Object.l [as queryRoute] (functionsWorker-0.10525428848427065.js:1344:13)
    at lQ (functionsWorker-0.10525428848427065.js:2217:13)
    at functionsWorker-0.10525428848427065.js:2297:17
    at n (functionsWorker-0.10525428848427065.js:41965:22)
```

Without source maps, it is difficult to track down the code responsible for the error.

I am using Cloudflare Pages with Remix. My Remix build script does generate source maps. But they don't match up with the file that `wrangler pages deploy` uploads. It seems to be offset by 50-60 lines.

Our log aggregator doesn't support uploading source maps, but if I was somehow able to access the real source maps, I could use a tool like https://github.com/mifi/stacktracify to manually do the mapping. 

It is hard to deploy code to production when you can't confidently track down the source of errors/bugs.

Thanks.

**Comment by BrandonNoad** [comment]
I was able to get the `.js` and `.js.map` files that are uploaded using `wrangler pages functions build --sourcemap --outdir .`. But again, the line numbers in the stacktrace are offset from the actual line in the file (e.g. line 36604 in the stacktrace and line 36654 in the .js file).

@mrbbot any idea what the cause of this offset could be?

**Reply by richardscarrott** [reply]
Hi @BrandonNoad did you figure out a nice solution to this?

We ended up using [advanced mode](https://developers.cloudflare.com/pages/functions/advanced-mode/) and telling wrangler not to bundle `wrangler pages deploy dist --no-bundle`, that way the code Remix generates it the code that's ultimately executed.

In the case of Remix, the wrangler bundle is kinda redundant as obviously Remix is doing the routing so you don't need the overhead of CF Pages routing (`/functions/[[path]].js`) on top.

It's just annoying having to rejig the [Remix cloudflare-pages template](https://github.com/remix-run/remix/tree/015ca5d567b509bf74dcdd416d8f2f114b34bb6f/templates/cloudflare-pages) every time we start a new project + still no source maps during development 🤯

**Comment by BrandonNoad** [comment]
@richardscarrott No, I still haven't found a solution for this issue.

It makes me wonder if anyone is actually using Remix + Cloudflare Pages in a real, production app 🤔 . Figuring out source maps should be a high-priority item IMO. As I mentioned before, **it is hard to deploy code to production when you can't confidently track down the source of errors/bugs**.

Thanks for the hint about Advanced Mode. I'll check it out and see if that will work for us. Would you be able to share the changes you've made to the default template?

**Comment by ShravanSunder** [comment]
The sourcemaps in a normal worker, vscode and attaching to wrangler dev is misaligned.  The line numbers do not match.

**Comment by BrandonNoad** [comment]
Looks like https://github.com/cloudflare/workers-sdk/pull/5861 may help with this issue
